tests/e2e/tests/conftest.py

```diff
@@ -32,3 +34,50 @@ def h() -> harness.Harness:
 
     LOG.debug("Cleanup")
     h.cleanup()
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "node_count: Mark a test to specify how many instance nodes need to be created",
+    )
+
+
+@pytest.fixture(scope="function")
+def node_count(request) -> int:
+    node_count_marker = request.node.get_closest_marker("node_count")
+    if not node_count_marker:
+        return 1
+    node_count_arg, *_ = node_count_marker.args
+    return int(node_count_arg)
+
+
+@pytest.fixture(scope="function")
+def instances(
+    h: harness.Harness, node_count: int, tmp_path: Path
+) -> List[harness.Instance]:
```

Comment:
If the function yields should it's return type indicate that it's a generator?

Reply:
hmm... yeah i suppose that's more clear 

in this case it's wrapped by a  `pytest.fixture` which will do things to the yielded value.  It ends up looking more like a contextmanager